example/src/main/scala/example/Example.scala

```diff
@@ -0,0 +1,66 @@
+package example
+
+import cats.Monad
+import cats.syntax.flatMap._
+import slog.{LoggerFactory, LoggingContext}
+
+class Example[F[_]](loggerFactory: LoggerFactory[F],
+                    loggingContext: LoggingContext[F])(implicit F: Monad[F]) {
+  private[this] val logger = loggerFactory.make[Example[F]]
+  // or
+  // private [this] val logger = loggerFactory.make("my-logger")
+
+  // example output json does not include timestamps, log levels, thread name,
+  // logger names etc to keep things simple. They are of course included in the real
+  // output.
+  def foo: F[Unit] = {
+    // simple log message. Might take from scope.
+    // {"message": "Hello world!!", "file": "Example.scala", "line": 19}
+    logger.info.msg("Hello world!!") >>
+      // Provide extract context to the log message
+      // {"message": "Hello", "file": "Example.scala", "line": 22, "correlation_id": "<VALUE>"}
+      logger.info.context("correlation_id", "<VALUE>").msg("Hello") >>
+      // Provide extra context and an exception
+      // {"message": "Hello", "file": "Example.scala", "line": 25, "correlation_id": "<VALUE>", "stack_trace": "Exception ..."}
+      logger.info
+        .context("correlation_id", "<VALUE>")
+        .msg(new Exception)("Hello") >>
+      loggingContext
+        .use("correlation_id", "<VALUE>")
+        .use("request_id", "<VALUE>")
+        .scoped {
+          // {"message": "test", "file": "Example.scala", "line": 33, "correlation_id": "<VALUE>", "request_id": "<VALUE>", "stack_trace": "Exception ..."}
+          logger.debug.msg(new Exception)("test") >>
+          // {"message": "test2", "file": "Example.scala", "line": 35, "correlation_id": "<VALUE>", "request_id": "<VALUE>"}
+          logger.info.msg("test2")
+        }
```

Comment:
here I try to illustrate how I would like the API to change
```scala
    logger.info("Hello world!!") >>
      // Provide extract context to the log message
      // {"message": "Hello", "file": "Example.scala", "line": 22, "correlation_id": "<VALUE>"}
      logger.context("correlation_id", "<VALUE>").info("Hello") >>
      // Provide extra context and an exception
      // {"message": "Hello", "file": "Example.scala", "line": 25, "correlation_id": "<VALUE>", "stack_trace": "Exception ..."}
      logger
        .context("correlation_id", "<VALUE>")
        .info(new Exception)("Hello") >>
      logger
        .context("correlation_id", "<VALUE>")
        .context("request_id", "<VALUE>")
        .scoped {
          // {"message": "test", "file": "Example.scala", "line": 33, "correlation_id": "<VALUE>", "request_id": "<VALUE>", "stack_trace": "Exception ..."}
          logger.debug(new Exception)("test") >>
          // {"message": "test2", "file": "Example.scala", "line": 35, "correlation_id": "<VALUE>", "request_id": "<VALUE>"}
          logger.info("test2") 
```

Reply:
I'm OK with switching `context` and log level. What I'm not OK is usage of `scoped` directly on `logger`. This somehow suggest that it is scoped only for *that particular logger*, which is not the case as it is scoped to the all loggers.

Reply:
How about calling it just `scope` instead of `scoped`? Meaning "I am creating a scope...".

Reply:
> scoped only for _that particular logger_

I mean, it could be the case. It depends on the other Loggers, if they share the same underlying "TaskLocal", and that really depends on how the programmer sets up the application

Reply:
Which is exactly why I want it to be separated. Because implementation should not really change semantics of the context.

Reply:
I'm not sure agree, I think this should be driven by how programmer wires it up -- which usually would be just 1 TaskLocal

Reply:
@hanny24 Good point about how calling scope (on logger instance) affects all loggers. Natural solution seems to have `object LoggingContext`. I see that it is `trait` passed as parameter, how do you propose multiple LoggingContexts would interect?